### DISCUSSION TITLE
does it support non-interactive way?

### DISCUSSION THREAD

**Discussion by swishcloud** [daniele77/cli#116] [Q&A] [answered]

**Comment by daniele77** [comment, marked as answer]
Hi @swishcloud .
What do you mean, exactly?
`cli` library supports a "non-interactive" mode by taking input from a command file: see the example that comes with the library `examples/filesession.cpp`.

**Reply by swishcloud** [reply]
such as GIT, it exit every time finished a command

**Reply by daniele77** [reply]
If you want you can add an exit instruction after each command handler, but I'm afraid `cli` library is not meant to work in that way.

**Reply by swishcloud** [reply]
got, thanks!
